ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/radiobutton/v1/radiobutton/clientlibs/site/js/radiobuttonview.js

```diff
@@ -108,19 +108,17 @@
         }
 
         updateValue(modelValue) {
-            if(modelValue != null) {
-                this.widget.forEach(widget => {
-                    if (widget.value != null && modelValue.toString() == widget.value.toString()) {
-                        widget.checked = true;
-                        widget.setAttribute(FormView.Constants.HTML_ATTRS.CHECKED, FormView.Constants.HTML_ATTRS.CHECKED);
-                        widget.setAttribute(FormView.Constants.ARIA_CHECKED, true);
-                    } else {
-                        widget.checked = false;
-                        widget.removeAttribute(FormView.Constants.HTML_ATTRS.CHECKED);
-                        widget.setAttribute(FormView.Constants.ARIA_CHECKED, false);
-                    }
-                }, this)
-            }
+            this.widget.forEach(widget => {
+                if (modelValue && widget.value != null && modelValue.toString() == widget.value.toString()) {
```

Comment:
Why have you moved this check ?

Reply:
if model value is different from widget value, how would the widget get updated to the new value?